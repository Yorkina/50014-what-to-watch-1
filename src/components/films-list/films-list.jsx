import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import FilmCard from '../film-card/film-card.jsx';


export const FilmsList = (props) => {
  const {films} = props;

  return (
    <div
      className="catalog__movies-list"
    >
      {films.map((film, ind) => <FilmCard
        key={ind}
        card={film}
      />)}
    </div>
  );
};

const filterFilms = (state) => {
  const {films, activeItem} = state;

  if (activeItem === `All Genre`) {
    return films;
  }

  return films.filter(({genre}) => genre === activeItem);
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  films: filterFilms(state),
});

export default connect(mapStateToProps)(FilmsList);

FilmsList.propTypes = {
  films: PropTypes.arrayOf(PropTypes.shape({
    href: PropTypes.string.isRequred,
    src: PropTypes.string.isRequred,
    title: PropTypes.string.isRequred,
  }))
};
