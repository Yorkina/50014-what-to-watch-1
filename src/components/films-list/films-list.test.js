import React from 'react';
import renderer from 'react-test-renderer';
import {FilmsList} from './films-list.jsx';


describe(`FilmsList`, () => {
  it(`should render`, () => {
    const tree = renderer
      .create(<FilmsList
        films={[{
          href: `fantastic-beasts-the-crimes-of-grindelwald.html`,
          previewImageSrc: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
          name: `Fantastic Beasts: The Crimes of Grindelwald`,
        }]}
      />).toJSON();

    expect(tree).toMatchSnapshot();

  });
});
