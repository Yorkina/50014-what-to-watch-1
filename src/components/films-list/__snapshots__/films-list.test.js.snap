// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsList should render 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        loop={true}
        muted={true}
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="fantastic-beasts-the-crimes-of-grindelwald.html"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
</div>
`;
